Web Inspector: Remove setting to toggle fuzzy matching for CSS autocompletion

rcaliman-apple opened the pull request:
#### b19502d24a56abeb148b282110659e47493d7a56
<pre>
Web Inspector: Remove setting to toggle fuzzy matching for CSS autocompletion
<a href="https://bugs.webkit.org/show_bug.cgi?id=241561">https://bugs.webkit.org/show_bug.cgi?id=241561</a>

Reviewed by Patrick Angle.

The fuzzy matching CSS autocompletion feature has been enabled by default for a few STP releases already.
No regressions for the autocompletion functionality have been uncovered since.
It&apos;s time to remove the setting that allows toggling off the feature.

* LayoutTests/inspector/unit-tests/css-keyword-completions.html:

Turns out the test was never exercising the fuzzy matching branch of autocompletion.
Becaue the `useFuzzy` flag was missing, it was falling back to testing prefix autocompletion.
The difference is that completion results are not plain strings, but `QueryResult` instances.

* Source/WebInspectorUI/Localizations/en.lproj/localizedStrings.js:
* Source/WebInspectorUI/UserInterface/Base/Setting.js:
* Source/WebInspectorUI/UserInterface/Models/CSSCompletions.js:
(WI.CSSCompletions.getCompletionText):

Moved the method to get the string value of a completion from `WI.CompletionSuggestionsView`
to be a static member of `WI.CSSCompletions`.

* Source/WebInspectorUI/UserInterface/Models/CSSKeywordCompletions.js:
(WI.CSSKeywordCompletions.forPartialPropertyName):
* Source/WebInspectorUI/UserInterface/Views/CompletionSuggestionsView.js:
(WI.CompletionSuggestionsView.prototype.set selectedIndex):
(WI.CompletionSuggestionsView.prototype.getCompletionText): Deleted.
* Source/WebInspectorUI/UserInterface/Views/SettingsTabContentView.js:
* Source/WebInspectorUI/UserInterface/Views/SpreadsheetTextField.js:
(WI.SpreadsheetTextField.prototype._updateCompletions):

Canonical link: <a href="https://commits.webkit.org/251564@main">https://commits.webkit.org/251564@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@295559">https://svn.webkit.org/repository/webkit/trunk@295559</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

rcaliman-apple commented on Source/WebInspectorUI/UserInterface/Models/CSSKeywordCompletions.js:
```diff
@@ -42,16 +42,10 @@ WI.CSSKeywordCompletions.forPartialPropertyName = function(text, {caretPosition,
-    if (useFuzzyMatching)
-        completions = WI.cssManager.propertyNameCompletions.executeQuery(text);
-    else
-        completions = WI.cssManager.propertyNameCompletions.startsWith(text);
```
Not removing the branch for prefix search via `CSSPropertyNameCompletions.startsWith()` because it is still used by CodeMirror editor instances in the Sources panel via `WI.CodeMirrorCompletionController`.

Ditto for `CSSCompletions.startsWith()`

patrickangle commented on Source/WebInspectorUI/UserInterface/Views/SettingsTabContentView.js:
```diff
@@ -391,7 +391,6 @@ WI.SettingsTabContentView = class SettingsTabContentView extends WI.TabContentVi
             let stylesGroup = experimentalSettingsView.addGroup(WI.UIString("Styles:"));
             stylesGroup.addSetting(WI.settings.experimentalEnableStylesJumpToEffective, WI.UIString("Show jump to effective property button"));
             stylesGroup.addSetting(WI.settings.experimentalEnableStylesJumpToVariableDeclaration, WI.UIString("Show jump to variable declaration button"));
-            stylesGroup.addSetting(WI.settings.experimentalCSSCompletionFuzzyMatching, WI.UIString("Use fuzzy matching for completion suggestions"));
```
Should this patch also remove `experimentalCSSCompletionFuzzyMatching` from WI.Settings?

rcaliman-apple replied:
Yes! Good catch.

kkinnunen-apple committed:
GPUP CGContexts create unattributed Metal resources
https://bugs.webkit.org/show_bug.cgi?id=241509
rdar://problem/94803367

Patch by Kimmo Kinnunen <kkinnunen@apple.com> on 2022-06-14
Reviewed by Dean Jackson.

Unattributed Metal related memory allocations grow GPUP footprint,
causing the process go over its memory limit.

Add a CoreGraphics SPI to mark that the CGContext resources belong to a
particular process.

Integrate this via IOSurface::setOwnershipIdentity(), as that is where
the current CGContext for IOSurfaces are instantiated and held.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/PAL/pal/spi/cg/CoreGraphicsSPI.h:
* Source/WebCore/PAL/pal/cg/CoreGraphicsSoftLink.cpp: Added.
* Source/WebCore/PAL/pal/cg/CoreGraphicsSoftLink.h: Added.
* Source/WebCore/PAL/pal/PlatformMac.cmake:
* Source/WebCore/platform/graphics/cocoa/IOSurface.h:
* Source/WebCore/platform/graphics/cocoa/IOSurface.mm:
(WebCore::IOSurface::ensurePlatformContext):
(WebCore::IOSurface::setOwnershipIdentity):

Canonical link: https://commits.webkit.org/251527@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295522 268f45cc-cd09-0410-ab3c-d52691b4dbfc

philn committed:
[BuildStream] Migrate to commit identifier for version keeping
https://bugs.webkit.org/show_bug.cgi?id=241469

Patch by Philippe Normand <philn@igalia.com> on 2022-06-14
Reviewed by Adrian Perez de Castro and Jonathan Bedard.

* Tools/buildstream/utils/generate-version:
* Tools/glib/svn-revision: Deleted.

Canonical link: https://commits.webkit.org/251528@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295523 268f45cc-cd09-0410-ab3c-d52691b4dbfc

youennf committed:
UserMediaCaptureManagerProxy should allow reusing any camera source of the same page
https://bugs.webkit.org/show_bug.cgi?id=241548
rdar://problem/95091899

Patch by Youenn Fablet <youennf@gmail.com> on 2022-06-14
Reviewed by Eric Carlson.

We are trying to reuse an existing camera source but we were only storing one camera source.
This does not work well in case the camera source gets cloned and then ended.
What we need to do is store all sources and if one is reusable, use it.

We use a WeakHashSet to store all camera sources (through getUserMedia or clone)
and iterate on them to either reuse and if not possible stop them.

This also ensures that a page does not have two live AVVideoCaptureSource sources.
If we had that, one source would interrupt the other and vice versa.

Fix a bug in which a cloned track would not have its PageIdentifier.

* Source/WebKit/UIProcess/Cocoa/UserMediaCaptureManagerProxy.cpp:
(WebKit::UserMediaCaptureManagerProxy::createCameraSource):
(WebKit::UserMediaCaptureManagerProxy::clone):
* Source/WebKit/UIProcess/Cocoa/UserMediaCaptureManagerProxy.h:
* Source/WebKit/UIProcess/Cocoa/UserMediaCaptureManagerProxy.messages.in:
* Source/WebKit/WebProcess/cocoa/RemoteRealtimeMediaSourceProxy.cpp:
(WebKit::RemoteRealtimeMediaSourceProxy::createRemoteCloneSource):
* Source/WebKit/WebProcess/cocoa/RemoteRealtimeMediaSourceProxy.h:
* Source/WebKit/WebProcess/cocoa/RemoteRealtimeVideoSource.cpp:
(WebKit::RemoteRealtimeVideoSource::clone):

Canonical link: https://commits.webkit.org/251529@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295524 268f45cc-cd09-0410-ab3c-d52691b4dbfc

clopez committed:
Port to Python3
https://bugs.webkit.org/show_bug.cgi?id=230098
<rdar://problem/82925374>

Reviewed by Aakash Jain.

Port browserperfdash-benchmark to python3 and invoke it with
the python3 interpreter on the CI.

* Tools/CISupport/build-webkit-org/steps.py:
(RunBenchmarkTests):
* Tools/Scripts/browserperfdash-benchmark:
* Tools/Scripts/webkitpy/browserperfdash/browserperfdash_runner.py:
(BrowserPerfDashRunner._parse_config_file):
(BrowserPerfDashRunner._upload_result):

Canonical link: https://commits.webkit.org/251530@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295525 268f45cc-cd09-0410-ab3c-d52691b4dbfc

philn committed:
[GLib] Support for JS scripts evaluation involving Promises
https://bugs.webkit.org/show_bug.cgi?id=241418

Patch by Philippe Normand <philn@igalia.com> on 2022-06-14
Reviewed by Adrian Perez de Castro and Michael Catanzaro.

The current APIs for `webkit_web_view_run_javascript_*` do not handle scripts involving Promises.
This new API fills the gap.

* Source/JavaScriptCore/API/glib/JSCDefines.h:
* Source/WebKit/Shared/API/APISerializedScriptValue.h:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/API/glib/APISerializedScriptValueGLib.cpp: Added.
(API::SharedJSContext::SharedJSContext):
(API::SharedJSContext::ensureContext):
(API::SharedJSContext::releaseContextIfNecessary):
(API::sharedContext):
(API::valueFromGVariant):
(API::coreValueFromGVariant):
(API::SerializedScriptValue::createFromGVariant):
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkit_web_view_run_javascript_in_world):
(webkit_web_view_run_async_javascript_function_in_world):
* Source/WebKit/UIProcess/API/gtk/WebKitWebView.h:
* Source/WebKit/UIProcess/API/wpe/WebKitWebView.h:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitWebView.cpp:
(testWebViewRunAsyncFunctions):
(beforeAll):
* Tools/TestWebKitAPI/glib/WebKitGLib/WebViewTest.cpp:
(WebViewTest::runJavaScriptInWorldAndWaitUntilFinished):
(WebViewTest::runAsyncJavaScriptFunctionInWorldAndWaitUntilFinished):
* Tools/TestWebKitAPI/glib/WebKitGLib/WebViewTest.h:

Canonical link: https://commits.webkit.org/251531@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295526 268f45cc-cd09-0410-ab3c-d52691b4dbfc

Smackteo committed:
[Gardening]REGRESSION(r289495): [ Monterey arm64 ] model-element/model-element-ready.html is a flaky assertion failure
https://bugs.webkit.org/show_bug.cgi?id=236678

Unreviewed test gardening.

* LayoutTests/platform/mac-wk2/TestExpectations:

Canonical link: https://commits.webkit.org/251532@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295527 268f45cc-cd09-0410-ab3c-d52691b4dbfc

pvollan committed:
Adopt SPI to get default content of Launch Services database
https://bugs.webkit.org/show_bug.cgi?id=241286
<rdar://64528683>

Reviewed by Geoffrey Garen.

Adopt SPI to get system content of Launch Services database. If this SPI is available,
it will replace the SPI we currently use to retrieve the database in the Network process.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebKit/NetworkProcess/cocoa/LaunchServicesDatabaseObserver.mm:
(WebKit::LaunchServicesDatabaseObserver::LaunchServicesDatabaseObserver):
(WebKit::LaunchServicesDatabaseObserver::startObserving):
(WebKit::LaunchServicesDatabaseObserver::~LaunchServicesDatabaseObserver):

Canonical link: https://commits.webkit.org/251533@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295528 268f45cc-cd09-0410-ab3c-d52691b4dbfc

Smackteo committed:
[Gardening][ iOS ] editing/pasteboard/dom-paste/dom-paste-same-origin.html is a flaky timeout/failure.
https://bugs.webkit.org/show_bug.cgi?id=237174

Unreviewed test gardening.

* LayoutTests/platform/ios/TestExpectations:

Canonical link: https://commits.webkit.org/251534@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295529 268f45cc-cd09-0410-ab3c-d52691b4dbfc

pvollan committed:
Create sandbox extension for Power logging service in the GPU process
https://bugs.webkit.org/show_bug.cgi?id=241227
<rdar://82442196>

Reviewed by Geoffrey Garen.

When power logging is enabled, the GPU process on macOS should be granted access to the Power logging service.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebCore/PAL/pal/spi/mac/PowerLogSPI.h: Copied from Source/WebKit/UIProcess/Cocoa/GPUProcessProxyCocoa.mm.
* Source/WebCore/platform/graphics/cg/ImageDecoderCG.cpp:
(WebCore::createImageSourceOptions):
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/GPUProcess/GPUProcess.h:
* Source/WebKit/GPUProcess/GPUProcess.messages.in:
* Source/WebKit/GPUProcess/mac/GPUProcessMac.mm:
(WebKit::GPUProcess::openDirectoryCacheInvalidated):
(WebKit::GPUProcess::enablePowerLogging):
* Source/WebKit/UIProcess/Cocoa/GPUProcessProxyCocoa.mm:
(WebKit::GPUProcessProxy::powerLoggingInTaskMode const):
(WebKit::GPUProcessProxy::enablePowerLogging):
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::WebProcessPool::registerNotificationObservers):
(WebKit::WebProcessPool::unregisterNotificationObservers):
* Source/WebKit/UIProcess/GPU/GPUProcessProxy.cpp:
(WebKit::GPUProcessProxy::didFinishLaunching):
* Source/WebKit/UIProcess/GPU/GPUProcessProxy.h:
* Source/WebKit/UIProcess/WebProcessPool.h:
* WebKit.xcworkspace/xcshareddata/xcschemes/All Source.xcscheme:

Canonical link: https://commits.webkit.org/251535@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295530 268f45cc-cd09-0410-ab3c-d52691b4dbfc

anttijk committed:
Nullptr crash in RenderLayerCompositor::requiresCompositingLayer
https://bugs.webkit.org/show_bug.cgi?id=241595
<rdar://88854150>

Reviewed by Simon Fraser.

* Source/WebCore/rendering/RenderLayerCompositor.cpp:
(WebCore::RenderLayerCompositor::requiresCompositingLayer const):

Add a null check. In -webkit-box-reflect case it is not clear renderer.layer() can't be null.

Canonical link: https://commits.webkit.org/251536@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295531 268f45cc-cd09-0410-ab3c-d52691b4dbfc

smfr committed:
Make RenderLayer ISOHeap allocated
https://bugs.webkit.org/show_bug.cgi?id=241606

Reviewed by Alan Bujtas.

Use ISOHeap allocation for RenderLayer. Losing the heap identifier capability is not a big loss;
RenderLayers themselves don't use much memory.

* Source/WebCore/rendering/RenderLayer.cpp:
* Source/WebCore/rendering/RenderLayer.h:

Canonical link: https://commits.webkit.org/251537@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295532 268f45cc-cd09-0410-ab3c-d52691b4dbfc

eric-carlson committed:
[Cocoa] Update camera list when system preferred camera changes
https://bugs.webkit.org/show_bug.cgi?id=241604
rdar://89694851

Reviewed by Youenn Fablet.

* Source/WebCore/platform/mediastream/mac/AVCaptureDeviceManager.mm:
(WebCore::AVCaptureDeviceManager::~AVCaptureDeviceManager):
(WebCore::AVCaptureDeviceManager::registerForDeviceNotifications):
(-[WebCoreAVCaptureDeviceManagerObserver observeValueForKeyPath:ofObject:change:context:]):

Canonical link: https://commits.webkit.org/251538@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295533 268f45cc-cd09-0410-ab3c-d52691b4dbfc

pxlcoder committed:
CrashTracer: MobileSafari at UIKitCore: -[UITextSearchingFindSession foundRange:forSearchString:inDocument:]
https://bugs.webkit.org/show_bug.cgi?id=241602
rdar://94622715

Reviewed by Wenson Hsieh.

UIKit retains the search string passed in to
-[UITextSearchingFindSession foundRange:forSearchString:inDocument:]. However,
since WebKit returns search results asynchronously, the string may be released
by the time we inform the aggregator of the results.

To fix, retain the search string in the capture list of the search completion
handler.

* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView performTextSearchWithQueryString:usingOptions:resultAggregator:]):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPage.mm:

Updated tests to use find interaction API rather than SPI.

Explicitly declare WKWebView's conformance to UITextSearching, since the
conformance to _UITextSearching in WKWebViewPrivate.h hides it.

_UITextSearching is still kept around for SPI clients, and will be removed
once they have migrated to the API.

(-[TestSearchAggregator foundRange:forSearchString:inDocument:]):
(-[TestSearchAggregator invalidateFoundRange:inDocument:]):
(testPerformTextSearchWithQueryStringInWebView):
(textRangesForQueryString):
(TEST):

Added an API test to verify the crash no longer occurs.

* Tools/TestWebKitAPI/ios/UIKitSPI.h:

Declared UITextSearchOptions properties as readwrite for use in testing.

Canonical link: https://commits.webkit.org/251539@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295534 268f45cc-cd09-0410-ab3c-d52691b4dbfc

nt1m committed:
Enable CSSTextAlignLast setting by default
https://bugs.webkit.org/show_bug.cgi?id=241241
<rdar://problem/94313487 >

Reviewed by Alan Bujtas.

All the WPT are on parity with Chrome & Firefox, so this should be ready to ship.

* Source/WTF/Scripts/Preferences/WebPreferencesExperimental.yaml:

Canonical link: https://commits.webkit.org/251540@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295535 268f45cc-cd09-0410-ab3c-d52691b4dbfc

stwrt committed:
[libpas] Add documentation on running specific tests in libpas
https://bugs.webkit.org/show_bug.cgi?id=241610

Reviewed by Mark Lam.

Add documentation on running specific libpas test cases.

* Source/bmalloc/libpas/ReadMe.md:

Canonical link: https://commits.webkit.org/251541@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295536 268f45cc-cd09-0410-ab3c-d52691b4dbfc

mwyrzykowski committed:
Update context menu icon for Remove Background
https://bugs.webkit.org/show_bug.cgi?id=241467

Patch by Mike Wyrzykowski <mwyrzykowski@apple.com> on 2022-06-14
Reviewed by Dean Jackson and Wenson Hsieh.

Update the symbol names for Copy Subject and Remove Background.

* Source/WebKit/UIProcess/API/Cocoa/_WKElementAction.mm:
(+[_WKElementAction imageForElementActionType:]):
Copy Subject will use circle.dashed.rectangle.

* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView removeBackgroundMenu]):
Remove Background should use circle.rectangle.filled.pattern.diagonalline.

(-[WKContentView menuWithInlineAction:image:identifier:handler:]):
Add image parameter for vertical context menu.

(-[WKContentView appHighlightMenu]):
We could use the "quicknote" symbol here but I don't think it's correct.
I left it empty because I don't think app highlights menu can appear in the
larger vertical context menu.

Canonical link: https://commits.webkit.org/251542@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295537 268f45cc-cd09-0410-ab3c-d52691b4dbfc

mattwoodrow committed:
Treat the composite operator for the bottom mask layer as source-over.
https://bugs.webkit.org/show_bug.cgi?id=241541

Reviewed by Tim Nguyen.

* LayoutTests/TestExpectations:
* Source/WebCore/rendering/RenderBoxModelObject.cpp:
(WebCore::RenderBoxModelObject::paintFillLayerExtended):
* Source/WebCore/rendering/style/FillLayer.h:
(WebCore::FillLayer::compositeForPainting const):

Canonical link: https://commits.webkit.org/251543@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295538 268f45cc-cd09-0410-ab3c-d52691b4dbfc

heycam committed:
Track SVG renderer updates using a NodeFlag instead of on Style::Update
https://bugs.webkit.org/show_bug.cgi?id=241489
<rdar://problem/94756741>

Reviewed by Antti Koivisto.

SVG renderer updates are currently tracked as a kind of Style::Update change.
When a page is updating attributes on many SVG elements, but is not making
changes that require a restyle on those elements, we can spend a lot of time
hashing to store and look up the Style::Update associated with an element.

This patch moves the "SVG renderer update is needed" state to a Node flag
instead, but continues to use the Style::Update mechanism for root tracking,
to reduce this overhead.

This is a 1% improvement on the MotionMark Suits sub-test.

* Source/WebCore/rendering/updating/RenderTreeUpdater.cpp:
(WebCore::RenderTreeUpdater::updateRenderTree):
(WebCore::RenderTreeUpdater::updateSVGRendererIfNeeded):
(WebCore::RenderTreeUpdater::updateElementRenderer):
* Source/WebCore/rendering/updating/RenderTreeUpdater.h:
* Source/WebCore/style/StyleUpdate.cpp:
(WebCore::Style::Update::addElement):
(WebCore::Style::Update::addSVGRendererUpdate):
* Source/WebCore/style/StyleUpdate.h:

Canonical link: https://commits.webkit.org/251544@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295539 268f45cc-cd09-0410-ab3c-d52691b4dbfc

JonWBedard committed:
[git-webkit] Automatically grant team access to private forks
https://bugs.webkit.org/show_bug.cgi?id=240819
<rdar://93780380>

Reviewed by Stephanie Lewis.

* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py:
(Tracker): Share ACCEPT_HEADER string.
(Tracker.credentials.validater): Ditto.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py:
(Setup.github): Allow provided team access to forked repository.
(Setup.git): Pass team for repo access from webkitscmpy.access config value.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/git_hub.py:
(GitHub): Share ACCEPT_HEADER string.
(GitHub.PRGenerator.create): Ditto.
(GitHub.PRGenerator.update): Ditto.
(GitHub.request): Ditto.
* metadata/git_config_extension: Define group to have access to apple/WebKit forks.

Canonical link: https://commits.webkit.org/251545@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295540 268f45cc-cd09-0410-ab3c-d52691b4dbfc

nt1m committed:
Remove redundant text-align-last BuilderCustom functions
https://bugs.webkit.org/show_bug.cgi?id=241607
<rdar://95131415>

Reviewed by Simon Fraser.

They might have been copy pasted from text-align, which needs them for the hasExplicitlySetTextAlign flag, which text-align-last does not have.

* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/style/StyleBuilderCustom.h:
(WebCore::Style::BuilderCustom::applyInitialTextAlignLast): Deleted.
(WebCore::Style::BuilderCustom::applyValueTextAlignLast): Deleted.

Canonical link: https://commits.webkit.org/251546@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295541 268f45cc-cd09-0410-ab3c-d52691b4dbfc

fujii committed:
TextureMapperLayer - avoiding multiple regions in paintUsingOverlapRegions
https://bugs.webkit.org/show_bug.cgi?id=140101

Reviewed by Don Olmstead.

paintUsingOverlapRegions has a code to merge nonOverlapRegion into
overlapRegion to avoid dividing into multiple small areas if
overlapRegion is bigger than nonOverlapRegion.

However, if the overlap region is just 1 pixel small from all the 4
side, this check would not catch that.

* LayoutTests/platform/glib/TestExpectations:
* Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp:
(WebCore::TextureMapperLayer::paintUsingOverlapRegions):
Compare totalArea of overlapRegion and nonOverlapRegion.

Canonical link: https://commits.webkit.org/251547@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295542 268f45cc-cd09-0410-ab3c-d52691b4dbfc

emw-apple committed:
Refactor built-product-archive to avoid deleting build products
https://bugs.webkit.org/show_bug.cgi?id=241586

Reviewed by Jonathan Bedard.

Deleting build products interferes with subsequent incremental builds.
Change the script to filter out what gets added to the archive, instead
of deleting from the directory being archived.

On Darwin-based platforms, this is done with mkbom and ditto's --bom
option, and surfaced in createZip via an `excludePatterns` argument.

Additionally, clean up the top-level logic before calling into
createZip, so that it doesn't need to know about the --minify option.

* Tools/CISupport/built-product-archive:

Canonical link: https://commits.webkit.org/251548@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295543 268f45cc-cd09-0410-ab3c-d52691b4dbfc

donny-dont committed:
[CMake] Specify some PlayStation WebKitRequirements components as optional
https://bugs.webkit.org/show_bug.cgi?id=241585

Reviewed by Ross Kirsling.

Explicitly mark the components of `find_package(WebKitRequirements)` as optional and required.
Some components can be optionally provided by WebKitRequirements or outside it as its own binary
so do a check for those cases, like libwebp.

* Source/cmake/OptionsPlayStation.cmake:

Canonical link: https://commits.webkit.org/251549@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295544 268f45cc-cd09-0410-ab3c-d52691b4dbfc

jyavenard committed:
Return error when parsing badly encoded std::variant IPC messages.
https://bugs.webkit.org/show_bug.cgi?id=241547
rdar://problem/95016858

Reviewed by Kimmo Kinnunen.

We check the variant's index at the start and insure its validity,
returning early of an invalid type index is found rather than asserting.

Code covered by making PasteboardCustomData::Entry use the std::variant encoder
and covered with ipc/pasteboard-write-custom-data.html

* Source/WebKit/Platform/IPC/ArgumentCoders.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder<PasteboardCustomData::Entry>::encode):
(IPC::ArgumentCoder<PasteboardCustomData::Entry>::decode):

Canonical link: https://commits.webkit.org/251550@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295545 268f45cc-cd09-0410-ab3c-d52691b4dbfc

patrickangle commented on Source/WebInspectorUI/UserInterface/Models/CSSCompletions.js:
```diff
@@ -131,6 +131,19 @@ WI.CSSCompletions = class CSSCompletions
 
+    static getCompletionText(completion)
+    {
+        console.assert(typeof completion === "string" || completion instanceof WI.QueryResult, completion);
+
+        if (typeof completion === "string")
+            return completion;
+
+        if (completion instanceof WI.QueryResult)
+            return completion.value;
+
+        return "";
```
Nit: You could just `console.assert(false, "explanation")` at the end instead of writing each condition twice. (I know you are just moving the code in this patch, but might be a good opportunity to simplify.

heycam committed:
Revert r295516 and r295373
https://bugs.webkit.org/show_bug.cgi?id=241621

Unreviewed, reverts:

r295516 "Make TextRun::isolatedCopy copy character data in more cases"
r295373 "Add a new DrawDecomposedGlyphs display list item to avoid repeatedly sending glyphs when using the GlyphDisplayListCache"

Canonical link: https://commits.webkit.org/251551@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295546 268f45cc-cd09-0410-ab3c-d52691b4dbfc

robert-jenner committed:
Revert [251527@main] GPUP CGContexts create unattributed Metal resources
https://bugs.webkit.org/show_bug.cgi?id=241509
rdar://problem/94803367

Unreviewed build fix/ revert.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebCore/PAL/pal/PlatformMac.cmake:
* Source/WebCore/PAL/pal/cg/CoreGraphicsSoftLink.cpp: Removed.
* Source/WebCore/PAL/pal/cg/CoreGraphicsSoftLink.h: Removed.
* Source/WebCore/platform/graphics/cocoa/IOSurface.h:
* Source/WebCore/platform/graphics/cocoa/IOSurface.mm:
(WebCore::IOSurface::ensurePlatformContext):
(WebCore::IOSurface::setOwnershipIdentity):

Canonical link: https://commits.webkit.org/251552@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295547 268f45cc-cd09-0410-ab3c-d52691b4dbfc

whsieh committed:
[iOS 15+] editing/selection/ios/hide-selection-in-tiny-contenteditable.html is a flaky failure
https://bugs.webkit.org/show_bug.cgi?id=241611
rdar://82183980

Reviewed by Aditya Keerthi.

This test is intended to check whether the text editing caret is visible or hidden on iOS when focusing a 1px by 1px
editable container; in particular, we expect to see the caret show up after making the tiny contenteditable area much
larger. However, the horizontal offset of the caret is sometimes shifted by a small amount to the left during the final
step of the test, likely due to the fact that the web view might still be in an unstable state after the animated zoom
in and out. Unfortunately, I haven't been able to reproduce this locally, so I can't confirm that this theory.

To try and mitigate this with a speculative fix, we:

1.  Update the test so that it just checks whether or not the caret rect is empty during each step.
2.  Use `immediateZoomToScale` instead of `zoomToScale`, so that we don't end up in an unstable state after finishing
    the second part of the test (which zooms in and out).

* LayoutTests/editing/selection/ios/hide-selection-in-tiny-contenteditable-expected.txt:
* LayoutTests/editing/selection/ios/hide-selection-in-tiny-contenteditable.html:
* LayoutTests/platform/ios-wk2/TestExpectations:

Mark the test as passing.

Canonical link: https://commits.webkit.org/251553@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295548 268f45cc-cd09-0410-ab3c-d52691b4dbfc

jjgriego committed:
Wasm LLInt support does not build on arm32-softfp
https://bugs.webkit.org/show_bug.cgi?id=241592

Patch by Joseph Griego <jgriego@igalia.com> on 2022-06-14
Reviewed by Yusuke Suzuki.

Example build failure: https://build.webkit.org/#/builders/61/builds/12719

Additionally, fixing the obvious problem in `FPRInfo.h` will fix the build
but does not produce working wasm support. So, for now, disable the feature
if are building with the softfp ABI

* Source/WTF/wtf/PlatformEnable.h:

Canonical link: https://commits.webkit.org/251554@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295549 268f45cc-cd09-0410-ab3c-d52691b4dbfc

rkirsling committed:
Temporal.PlainTime#since should handle ceil/floor properly
https://bugs.webkit.org/show_bug.cgi?id=241623

Reviewed by Yusuke Suzuki.

This patch adds a spec step that our implementation was missing:

    DifferenceTemporalPlainTime ( operation, temporalTime, other, options )
        ...
        9. If operation is ~since~, then
            a. Set roundingMode to ! NegateTemporalRoundingMode(roundingMode).

* JSTests/test262/expectations.yaml:
Mark four test cases as passing.

* Source/JavaScriptCore/runtime/TemporalPlainTime.cpp:
(JSC::TemporalPlainTime::since const):

Canonical link: https://commits.webkit.org/251555@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295550 268f45cc-cd09-0410-ab3c-d52691b4dbfc

rkirsling committed:
Temporal.Instant#round should treat the Big Bang (not the Unix epoch) as zero
https://bugs.webkit.org/show_bug.cgi?id=241622

Reviewed by Yusuke Suzuki.

Implement the spec fix of tc39/proposal-temporal#2210.
This change is quite simple, as we can just update the Int128 version of roundNumberToIncrement.

* JSTests/test262/expectations.yaml:
Mark four test cases as passing.

* Source/JavaScriptCore/runtime/TemporalObject.cpp:
(JSC::roundNumberToIncrement):
(JSC::abs): Deleted.

Canonical link: https://commits.webkit.org/251556@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295551 268f45cc-cd09-0410-ab3c-d52691b4dbfc

jyavenard committed:
REGRESSION (iOS 16): <img> with video source fails to render—affects iCloud live photos (IOKit blocking)
https://bugs.webkit.org/show_bug.cgi?id=241589

Reviewed by Simon Fraser.

Stop using IOKit in the content process. Instead we create a SharedMemory backed
ShareableBitmap and send it to the content process. The content process takes
ownership of this ShareableBitmap.

Covered by new media/media-hevc-video-as-img.html test

* LayoutTests/media/content/bipbop-hevc.mp4: Added.
* LayoutTests/media/content/test-video-as-img.png: Added.
* LayoutTests/media/media-hevc-video-as-img-expected.html: Added.
* LayoutTests/media/media-hevc-video-as-img.html: Added.
* LayoutTests/platform/glib/TestExpectations:
* Source/WebKit/GPUProcess/media/RemoteImageDecoderAVFProxy.cpp:
(WebKit::RemoteImageDecoderAVFProxy::createFrameImageAtIndex):
* Source/WebKit/GPUProcess/media/RemoteImageDecoderAVFProxy.h:
* Source/WebKit/GPUProcess/media/RemoteImageDecoderAVFProxy.messages.in:
* Source/WebKit/WebProcess/GPU/media/RemoteImageDecoderAVF.cpp:
(WebKit::RemoteImageDecoderAVF::createFrameImageAtIndex):

Canonical link: https://commits.webkit.org/251557@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295552 268f45cc-cd09-0410-ab3c-d52691b4dbfc

twilco committed:
AX: AccessibilityObject::insertText does not insert any text into password fields
https://bugs.webkit.org/show_bug.cgi?id=241613

Reviewed by Chris Fleizach and Andres Gonzalez.

Prior to this patch, AccessibilityObject::insertText used to return
early if Element::shouldUseInputMethod returned false. We need to
exclude password fields from this check, which this patch implements.

* LayoutTests/accessibility/insert-text-into-password-field-expected.txt: Added.
* LayoutTests/accessibility/insert-text-into-password-field.html: Added.
* LayoutTests/platform/glib/TestExpectations: Disable new test.
* LayoutTests/platform/ios/TestExpectations: Enable new test.
* LayoutTests/platform/win/TestExpectations: Disable new test.
* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::insertText):

Canonical link: https://commits.webkit.org/251558@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295553 268f45cc-cd09-0410-ab3c-d52691b4dbfc

aperezdc committed:
[WPE][GTK] REGRESSION(r292084): Browser frontend host is missing recent additions
https://bugs.webkit.org/show_bug.cgi?id=241594

Reviewed by Patrick Angle.

Update BrowserInspectorFrontendHost.js to add missing methods, update
signatures of changed ones, and remove those which are no longer present
in the IDL interface. This makes the Web Inspector work again when
loaded over HTTP.

* Source/WebInspectorUI/UserInterface/Base/BrowserInspectorFrontendHost.js:
(window.InspectorFrontendHost.WI.BrowserInspectorFrontendHost.prototype.canLoad): Added, always
returns false.
(window.InspectorFrontendHost.WI.BrowserInspectorFrontendHost.prototype.load): Added stub.
(window.InspectorFrontendHost.WI.BrowserInspectorFrontendHost.prototype.getPath): Added stub.
(window.InspectorFrontendHost.WI.BrowserInspectorFrontendHost.prototype.canPickColorFromScreen):
Added, always returns false.
(window.InspectorFrontendHost.WI.BrowserInspectorFrontendHost.prototype.pickColorFromScreen):
Added stub.
(window.InspectorFrontendHost.WI.BrowserInspectorFrontendHost.prototype.getCurrentX): Added stub.
(window.InspectorFrontendHost.WI.BrowserInspectorFrontendHost.prototype.getCurrentY): Added stub.
(window.InspectorFrontendHost.WI.BrowserInspectorFrontendHost.prototype.setPath): Added stub.
(window.InspectorFrontendHost.WI.BrowserInspectorFrontendHost.prototype.get port): Deleted.
(window.InspectorFrontendHost.WI.BrowserInspectorFrontendHost.prototype.append): Deleted.
(window.InspectorFrontendHost.WI.BrowserInspectorFrontendHost.prototype.close): Deleted.

Canonical link: https://commits.webkit.org/251559@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295554 268f45cc-cd09-0410-ab3c-d52691b4dbfc

philn committed:
[GStreamer][WebRTC] canvas to PeerConnection improvements
https://bugs.webkit.org/show_bug.cgi?id=241330

Patch by Philippe Normand <philn@igalia.com> on 2022-06-15
Reviewed by Miguel Gomez and Xabier Rodriguez-Calvar.

Video frames exported from the WebGL canvas need to be flipped on GStreamer ports and their format
should be RGBA. Frames coming from 2D canvas don't need to be flipped and have BGRA format.

Texture flipping is now handled in the player which is useful when painting canvas to a video
element. And flipping is also now handled in RTC outgoing video sources, which is useful when
exporting a canvas to a WebRTC PeerConnection.

The 30fps framerate is hardcoded, ideally should be inferred from the compositor refresh rate.
Having a framerate > 1 is specially needed for the canvas->WebRTC use-case, otherwise a 1/1
framerate would be set in caps, making the corresponding video stream slower than the actual refresh
rate.

* LayoutTests/platform/glib/TestExpectations:
* Source/WebCore/html/HTMLCanvasElement.cpp:
(WebCore::HTMLCanvasElement::toVideoFrame):
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::MediaPlayerPrivateGStreamer::updateTextureMapperFlags):
* Source/WebCore/platform/graphics/gstreamer/VideoFrameGStreamer.cpp:
(WebCore::VideoFrameGStreamer::createFromPixelBuffer):
* Source/WebCore/platform/graphics/gstreamer/VideoFrameGStreamer.h:
* Source/WebCore/platform/graphics/texmap/GraphicsContextGLTextureMapper.cpp:
(WebCore::GraphicsContextGLTextureMapper::paintCompositedResultsToVideoFrame):
* Source/WebCore/platform/mediastream/gstreamer/MockRealtimeVideoSourceGStreamer.cpp:
(WebCore::MockRealtimeVideoSourceGStreamer::updateSampleBuffer):
* Source/WebCore/platform/mediastream/gstreamer/RealtimeOutgoingVideoSourceGStreamer.cpp:
(WebCore::RealtimeOutgoingVideoSourceGStreamer::RealtimeOutgoingVideoSourceGStreamer):
(WebCore::RealtimeOutgoingVideoSourceGStreamer::setPayloadType):
* Source/WebCore/platform/mediastream/gstreamer/RealtimeOutgoingVideoSourceGStreamer.h:

Canonical link: https://commits.webkit.org/251560@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295555 268f45cc-cd09-0410-ab3c-d52691b4dbfc

alanbaradlay committed:
REGRESSION (r289443): Page contents disappear after entering a letter in the username field of bmoharris.com
https://bugs.webkit.org/show_bug.cgi?id=241625
<rdar://93516876>

Reviewed by Simon Fraser.

overflow: clip forbids scrolling entirely, through any mechanism (https://drafts.csswg.org/css-overflow/#propdef-overflow)

* LayoutTests/fast/scrolling/scrollIntoView-with-overflow-clip-expected.html: Added.
* LayoutTests/fast/scrolling/scrollIntoView-with-overflow-clip.html: Added.
* LayoutTests/fast/scrolling/selection-reveal-with-overflow-clip-expected.html: Added.
* LayoutTests/fast/scrolling/selection-reveal-with-overflow-clip.html: Added.
* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::RenderLayer::allowsCurrentScroll const): Make sure the content is not scrollable when overflow: clip is set on the renderer.

Canonical link: https://commits.webkit.org/251561@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295556 268f45cc-cd09-0410-ab3c-d52691b4dbfc

kkinnunen-apple committed:
GPUP CGContexts create unattributed Metal resources
https://bugs.webkit.org/show_bug.cgi?id=241509
rdar://problem/94803367

Patch by Kimmo Kinnunen <kkinnunen@apple.com> on 2022-06-15
Reviewed by Dean Jackson.

Re-land the patch after revert with disabling the feature for iOS simulator
since task_identity_t is not available in the simulator.

Unattributed Metal related memory allocations grow GPUP footprint,
causing the process go over its memory limit.

Add a CoreGraphics SPI to mark that the CGContext resources belong to a
particular process.

Integrate this via IOSurface::setOwnershipIdentity(), as that is where
the current CGContext for IOSurfaces are instantiated and held.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/PAL/pal/spi/cg/CoreGraphicsSPI.h:
* Source/WebCore/PAL/pal/cg/CoreGraphicsSoftLink.cpp: Added.
* Source/WebCore/PAL/pal/cg/CoreGraphicsSoftLink.h: Added.
* Source/WebCore/PAL/pal/PlatformMac.cmake:
* Source/WebCore/platform/graphics/cocoa/IOSurface.h:
* Source/WebCore/platform/graphics/cocoa/IOSurface.mm:
(WebCore::IOSurface::ensurePlatformContext):
(WebCore::IOSurface::setOwnershipIdentity):

Canonical link: https://commits.webkit.org/251562@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295557 268f45cc-cd09-0410-ab3c-d52691b4dbfc

AndresGonzalezApple committed:
AX: CrashTracer: com.apple.WebKit.WebContent at com.apple.WebCore: WTF::RetainPtr<NSMutableArray> WTF::createNSArray<WTF::Vector<WTF::RefPtr<WebCore::AXCoreObject, WTF::RawPtrTraits<WebCore::AXCoreObject>, WTF::DefaultRefDerefTraits<WebC. web content
https://bugs.webkit.org/show_bug.cgi?id=241619

Reviewed by Chris Fleizach.

Check for nullptr before dereferencing the RefPtrs passed to makeNSArray. In some cases, the given AccessibilityChildrenVector may contain nullptrs.

* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperBase.mm:
(makeNSArray):

Canonical link: https://commits.webkit.org/251563@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295558 268f45cc-cd09-0410-ab3c-d52691b4dbfc

rcaliman-apple committed:
Web Inspector: Remove setting to toggle fuzzy matching for CSS autocompletion
https://bugs.webkit.org/show_bug.cgi?id=241561

Reviewed by Patrick Angle.

The fuzzy matching CSS autocompletion feature has been enabled by default for a few STP releases already.
No regressions for the autocompletion functionality have been uncovered since.
It's time to remove the setting that allows toggling off the feature.

* LayoutTests/inspector/unit-tests/css-keyword-completions.html:

Turns out the test was never exercising the fuzzy matching branch of autocompletion.
Becaue the `useFuzzy` flag was missing, it was falling back to testing prefix autocompletion.
The difference is that completion results are not plain strings, but `QueryResult` instances.

* Source/WebInspectorUI/Localizations/en.lproj/localizedStrings.js:
* Source/WebInspectorUI/UserInterface/Base/Setting.js:
* Source/WebInspectorUI/UserInterface/Models/CSSCompletions.js:
(WI.CSSCompletions.getCompletionText):

Moved the method to get the string value of a completion from `WI.CompletionSuggestionsView`
to be a static member of `WI.CSSCompletions`.

* Source/WebInspectorUI/UserInterface/Models/CSSKeywordCompletions.js:
(WI.CSSKeywordCompletions.forPartialPropertyName):
* Source/WebInspectorUI/UserInterface/Views/CompletionSuggestionsView.js:
(WI.CompletionSuggestionsView.prototype.set selectedIndex):
(WI.CompletionSuggestionsView.prototype.getCompletionText): Deleted.
* Source/WebInspectorUI/UserInterface/Views/SettingsTabContentView.js:
* Source/WebInspectorUI/UserInterface/Views/SpreadsheetTextField.js:
(WI.SpreadsheetTextField.prototype._updateCompletions):

Canonical link: https://commits.webkit.org/251564@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295559 268f45cc-cd09-0410-ab3c-d52691b4dbfc
